packages/node/src/integrations/anr/index.ts

```diff
@@ -0,0 +1,172 @@
+import type { Options, WorkerStartData } from './common';
+import { base64WorkerScript } from './worker-script';
+
+const DEFAULT_INTERVAL = 50;
+const DEFAULT_HANG_THRESHOLD = 5000;
+
+type WorkerNodeV14 = Worker & { new (filename: string | URL, options?: WorkerOptions): Worker };
+
+type WorkerThreads = {
+  Worker: WorkerNodeV14;
+};
+
+function log(message: string, ...args: unknown[]): void {
+  logger.log(`[ANR] ${message}`, ...args);
+}
+
+/**
+ * We need to use dynamicRequire because worker_threads is not available in node < v12 and webpack error will when
+ * targeting those versions
+ */
+function getWorkerThreads(): WorkerThreads {
+  return dynamicRequire(module, 'worker_threads');
+}
+
+/**
+ * Gets contexts by calling all event processors. This relies on being called after all integrations are setup
+ */
+async function getContexts(client: NodeClient): Promise<Contexts> {
+  let event: Event | null = { message: 'ANR' };
+  const eventHint: EventHint = {};
+
+  for (const processor of client.getEventProcessors()) {
+    if (event === null) break;
+    event = await processor(event, eventHint);
+  }
+
+  return event?.contexts || {};
+}
+
+interface InspectorApi {
+  open: (port: number) => void;
+  url: () => string | undefined;
+}
+
+/**
+ * Starts the node debugger and returns the inspector url.
+ *
+ * When inspector.url() returns undefined, it means the port is already in use so we try the next port.
+ */
+function startInspector(startPort: number = 9229): string | undefined {
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const inspector: InspectorApi = require('inspector');
+  let inspectorUrl: string | undefined = undefined;
+  let port = startPort;
+
+  while (inspectorUrl === undefined && port < startPort + 100) {
+    inspector.open(port);
+    inspectorUrl = inspector.url();
+    port++;
+  }
+
+  return inspectorUrl;
+}
+
+/**
+ * Starts a thread to detect App Not Responding (ANR) events
+ */
+export class Anr implements Integration {
+  public name: string = 'Anr';
+
+  public constructor(private readonly _options: Partial<Options> = {}) {}
+
+  /** @inheritdoc */
+  public setupOnce(): void {
+    //
+  }
+
+  /** @inheritdoc */
+  public setup(client: NodeClient): void {
+    if ((NODE_VERSION.major || 0) < 16) {
+      throw new Error('ANR detection requires Node 16 or later');
+    }
+
+    // setImmediate is used to ensure that all other integrations have been setup
+    setImmediate(() => this._startWorker(client));
+  }
+
+  /**
+   * Starts the ANR worker thread
+   */
+  private async _startWorker(client: NodeClient): Promise<void> {
+    const contexts = await getContexts(client);
+    const initOptions = client.getOptions();
+
+    const dsn = client.getDsn();
+
+    if (!dsn) {
+      return;
+    }
+
+    // These will not be inaccurate if sent later from the worker thread
+    delete contexts.app?.app_memory;
+    delete contexts.device?.free_memory;
+
```

Comment:
Setting to undefined is always better. For example, here's a StackOverflow link for comparing the performance: https://stackoverflow.com/questions/13982306/in-javascript-what-is-more-efficient-deleting-an-element-or-setting-it-to-undef